### PR TITLE
refactor: move outside lib/scheme/*.lib to buildin &'static str

### PR DESCRIPTION
**Pull request by CurryPseudo**
* refactor LibrarySearcher::search_lib's return type from file path to char stream
* move standard lib file to &'static str could:
    * avoid setting current path to ruschm or copying lib file to current path every time
    * ensure tests that involve standard library will not failed because of path or file IO problem

### CODE REVIEW

File: src/interpreter/library/mod.rs [hunk 1]
```diff
@@ -117,37 +121,35 @@ impl StandardLibrarySearcher {
...
-                    .join(library_name.extract_data().path())
-                    .with_extension("sld"),
-            ),
-            Err(_) => None,
-        }
+impl StandardLibrarySearcher {
+    pub fn get_include_library(library_name: &LibraryName) -> Option<&'static str> {
+        let defined_librarys = include::library_strs()
+            .into_iter()
+            .collect::<HashMap<_, _>>();
```
Comment (Danielmelody) [inline src/interpreter/library/mod.rs:128]: seems this collecting can be written in library_strs

File: src/interpreter/library/mod.rs [hunk 2]
```diff
@@ -117,37 +121,35 @@ impl StandardLibrarySearcher {
...
+            Some(library_str) => Some(Box::new(library_str.chars())),
+            None => match std::env::current_dir() {
+                Ok(pwd) => {
+                    let path = pwd
+                        .join("lib")
+                        .join(library_name.extract_data().path())
+                        .with_extension("sld");
+                    match file_char_stream(&path) {
+                        Ok(char_stream) => Some(Box::new(char_stream)),
+                        Err(_) => None,
```
Comment (Danielmelody) [inline src/interpreter/library/mod.rs:148]: Error information lost?

File: src/interpreter/library/mod.rs (hunk 1 above)
Comment (CurryPseudo) [reply src/interpreter/library/mod.rs:128]: I think that library_strs is just a list of include library, not a hashmap, usage in this function required a hash search, but in other place may just need to iterate over whole list

File: src/interpreter/library/mod.rs (hunk 2 above)
Comment (CurryPseudo) [reply src/interpreter/library/mod.rs:148]: This is a library searcher, if load char stream from file failed, treat it as a result of library not found, not an error.
Then it is useful to combine this StandardLibrarySearcher with other user defined searcher (implementation of file extension priority, etc).
If the io error information matters, maybe need a log, but not returned by Result<_, _>

File: src/interpreter/library/mod.rs (hunk 1 above)
Comment (Danielmelody) [reply src/interpreter/library/mod.rs:128]: So, in this single function, if you just wanna get lib from name over a vector, you should search it or binary search it from a sorted one.

Comment (Danielmelody) [reply src/interpreter/library/mod.rs:128]: Constructing a HashMap is an O(n)  complexity step with a big constant factor.

File: src/interpreter/library/mod.rs (hunk 2 above)
Comment (Danielmelody) [reply src/interpreter/library/mod.rs:148]: But you still need to place the "File Not Found" Error in somewhere, and those 
detailed information becomes a single assumption, still lost there.

File: src/interpreter/library/mod.rs [hunk 3]
```diff
@@ -117,37 +121,38 @@ impl StandardLibrarySearcher {
...
+                    let path = pwd
+                        .join("lib")
+                        .join(library_name.extract_data().path())
+                        .with_extension("sld");
+                    // file exist
+                    if path.exists() {
+                        // directly unwrap to panic for other io error
+                        Some(Box::new(file_char_stream(&path).unwrap()))
+                    } else {
+                        None
```
Comment (Danielmelody) [inline src/interpreter/library/mod.rs:151]: I don't see the panic...

**Comment by Danielmelody** [comment]
Part of the motivation isn't reasonable, instead of caching &'static str, we should cache the parsed AST.
LIBRARY_STRS is a dynamic status within an evaluation, should not be static.
Close it.

**Pull request closed without merging**